manifests/install.pp

```diff
@@ -1,63 +1,24 @@
-# **NOTE: THIS IS A [PRIVATE](https://github.com/puppetlabs/puppetlabs-stdlib#assert_private) CLASS**
-#
 # Install the IPTables and IP6Tables components
 #
-# This also installs fallback startup scripts that come into play should the
-# regular processes fail to start due to a race consition with DNS.
+# @param ipv4_package
+#   The package used to manage ipv4 rules
+#
+#   * Default from module data
+#
+# @param ipv6_package
+#   The package used to manage ipv6 rules
 #
-class iptables::install {
-  assert_private()
+#   * Default from module data
+#
+class iptables::install (
+  String[1] $ipv4_package,
+  String[1] $ipv6_package
+){
   simplib::assert_metadata($module_name)
 
-  # IPV4-only stuff
-  package { 'iptables': ensure => $iptables::ensure }
-
-  file { '/etc/init.d/iptables':
-    ensure  => 'file',
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0755',
-    content => file("${module_name}/iptables"),
-    seltype => 'iptables_initrc_exec_t'
-  }
-
-  # --------------------------------------------------
-  # Set the iptables startup script to fail safe.
-  #
-  file { '/etc/init.d/iptables-retry':
-    ensure  => 'file',
-    owner   => 'root',
-    group   => 'root',
-    mode    => '0744',
-    content => file("${module_name}/iptables-retry"),
-    seltype => 'iptables_initrc_exec_t'
-  }
-
-  Package['iptables'] -> File['/etc/init.d/iptables']
-  Package['iptables'] -> File['/etc/init.d/iptables-retry']
+  ensure_packages($ipv4_package, {'ensure' => $iptables::ensure})
```

Comment:
Since this is no longer a private class, there is no guarantee `$iptables::ensure exists`